File: abc-267/c.py
from typing import List


def create_cumulative_sum_list(nums: List[int]) -> List[int]:
    res, sum = [0], 0
    for num in nums:
        sum += num
        res.append(sum)
    return res


def main():
    n, m = map(int, input().split())
    a_list = list(map(int, input().split()))
    cumulative_sum_a_list = create_cumulative_sum_list(a_list)
    ans, x = 0, 0

    for i in range(n - m + 1):
        if i == 0:
            for j in range(m):
                x += (j + 1) * a_list[j]
                ans = x
            continue
        x -= cumulative_sum_a_list[i+m-1] - cumulative_sum_a_list[i-1]
        x += m * a_list[i+m-1]
        ans = max(ans, x)

    print(ans)


if __name__ == '__main__':
    main()

File: abc-267/c.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

var sc = bufio.NewScanner(os.Stdin)
var rdr = bufio.NewReaderSize(os.Stdin, 1000000)

func stringInput() string {
	sc.Scan()
	return sc.Text()
}

func intInput() int {
	sc.Scan()
	x, err := strconv.Atoi(sc.Text())
	if err != nil {
		panic(err)
	}
	return x
}

func intSliceInput(n int) []int {
	sl := []int{}
	for i := 0; i < n; i++ {
		x := intInput()
		sl = append(sl, x)
	}
	return sl
}

func min(x, y int) int {
	if x < y {
		return x
	}
	return y
}

func max(x, y int) int {
	if x > y {
		return x
	}
	return y
}

func initSlSl(y, x int) [][]int {
	sl := make([][]int, y)
	if x == -1 {
		for i := 0; i < y; i++ {
			sl[i] = []int{}
		}
		return sl
	}
	for i := 0; i < y; i++ {
		sl[i] = make([]int, x)
	}
	return sl
}

func init() {
	sc.Split(bufio.ScanWords)
}

func createCumulativeSumSl(sl []int) []int {
	res := []int{0}
	sum := 0
	for _, num := range sl {
		sum += num
		res = append(res, sum)
	}
	return res
}

func main() {
	n, m := intInput(), intInput()
	aSl := intSliceInput(n)
	cumulativeSumASl := createCumulativeSumSl(aSl)
	ans, x := 0, 0

	for i := 0; i < n-m+1; i++ {
		if i == 0 {
			for j := 0; j < m; j++ {
				x += (j + 1) * aSl[j]
				ans = x
			}
			continue
		}
		x -= cumulativeSumASl[i+m-1] - cumulativeSumASl[i-1]
		x += m * aSl[i+m-1]
		ans = max(x, ans)
	}
	fmt.Println(ans)
}
